docs/machine-requirements.md

```diff
@@ -2,20 +2,51 @@
 
 Whether you want to contribute to Aspire, or just use the latest build of Aspire, these are the common pieces you need to install.
 
-## (Windows) Install Visual Studio
+## (Windows) Visual Studio
 
 On Windows, Visual Studio contains special tooling support for .NET Aspire that you will want to have.
 
+### Install Visual Studio
+
 [Visual Studio 2022 version 17.9 Preview](https://visualstudio.microsoft.com/vs/preview/)
 
 When you install, ensure that both:
 * `ASP.NET and web development` Workload is checked.
 * `.NET Aspire SDK` component in `Individual components` is checked.
 
-## Install the latest .NET 8 SDK
+### Install the latest .NET 8 SDK
+
 [.NET 8 SDK](https://github.com/dotnet/installer#installers-and-binaries)
 
-## Install Docker Desktop
+### Install Docker Desktop
+
 * [Windows](https://docs.docker.com/desktop/install/windows-install/)
 * [MacOS X](https://docs.docker.com/desktop/install/mac-install/)
 * [Linux](https://docs.docker.com/desktop/install/linux-install/)
+
+## (Windows / Linux / Mac) DevContainer in VSCode
+
+On Windows you could also use VSCode with the DevContainers extension. This requieres that you have installed a container engine installed.
+Currently it's only tested with Docker Desktop.
+
+> Warning: This will use around 16GB of RAM, after you loaded the solution.
+
+### Install VSCode with DevContainers Extension
+
+* [VSCode](https://code.visualstudio.com/Download)
+* [DevContainers Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
+
+### Install Docker Desktop
+
+* [Windows](https://docs.docker.com/desktop/install/windows-install/)
+* [MacOS X](https://docs.docker.com/desktop/install/mac-install/)
+* [Linux](https://docs.docker.com/desktop/install/linux-install/)
+
+## (Browser) Codespaces
+
+Just start the Codespaces in your fork. The initialisation of the code space takes around 5 mins. After that you can open the solution.
+This will take on the free version of Codesapce around 10 mins.
+
+> Warning: With the free version of Codespaces the development experience is not nice. We recommend using at least a Codespace with 16GB of RAM or use your local VSCode / DevContainers instance.
```

Comment:
I think there is a setting somewhere that I can use to increase this (billed to the .NET foundation..)

Reply:
I don't know how it works for members of the .NET Foundation or Microsoft employees, but for the rest of the world you need to pay a bill if you want to use it.

That's why I personally use the local dev containers for repositories like that more often.

Reply:
![image](https://github.com/dotnet/aspire/assets/11291885/48733ba9-b875-43fb-a5c4-336b0992fa44)
This is how you can configure. because It don't want to invest money I didn't try what will happen if you click create :D

Reply:
If I remember right, we can simply add as dotnet org members folks like you who want to use this. Then the foundation pays.